File: frontend/src/services/groupService.js
import api from './api';

export const groupService = {
  getAllGroups: async () => {
    const response = await api.get('/groups');
    return response.data;
  },

  getGroupById: async (id) => {
    const response = await api.get(`/groups/${id}`);
    return response.data;
  },

  createGroup: async (groupData) => {
    const response = await api.post('/groups', groupData);
    return response.data;
  },

  updateGroup: async (id, groupData) => {
    const response = await api.put(`/groups/${id}`, groupData);
    return response.data;
  },

  deleteGroup: async (id) => {
    await api.delete(`/groups/${id}`);
  },
};

File: backend/Nexus-IMS/src/main/java/com/learn/nexus_ims/services/GroupService.java
package com.learn.nexus_ims.services;

import java.util.List;

import com.learn.nexus_ims.dtos.GroupDto;

public interface GroupService {
    GroupDto createGroup(GroupDto groupDto);
    GroupDto updateGroup(Long groupId, GroupDto groupDto);
    void deleteGroup(Long groupId);
    List<GroupDto> getAllGroups();
    GroupDto getGroupById(Long groupId);
}
